overrides/cf.d.ts

```diff
@@ -289,97 +297,693 @@ interface RequestInitCfPropertiesImageMinify {
 }
 
 /**
- * In addition to the properties on the standard Request object,
- * the cf object contains extra information about the request provided
- * by Cloudflare's edge.
- *
- * Note: Currently, settings in the cf object cannot be accessed in the
- * playground.
+ * Request metadata provided by Cloudflare's edge.
  */
-interface IncomingRequestCfProperties {
+type IncomingRequestCfProperties = {
```

Comment:
OOC - why change from interface to type

Reply:
Because of the intersection with the union types at the bottom.

Reply:
I don't see any intersection?

Reply:
Here on [line 422](https://github.com/cloudflare/workers-types/pull/301/files#diff-434e5dced8c8102a0e01720275954a43e10c6fb3767ffa9728ad78845c94a916R422)

Reply:
I've moved it up to the start of the type so it's more clear :)

Reply:
OK so the way I would go about this would be:

```ts
interface BaseIncomingRequestCfProperties { ... }
interface T1IncomingRequestCfProperties extends BaseIncomingRequestCfProperties {
  country: "T1"
}
interface OtherIncomingRequestCfProperties extends BaseIncomingRequestCfProperties {
 ...
}
type IncomingRequestCfProperties = T1IncomingRequestCfProperties | OtherIncomingRequestCfProperties;
```